tokio/src/process/mod.rs

```diff
@@ -1190,6 +1192,41 @@ impl Child {
     ///     }
     /// }
     /// ```
+    ///
+    /// You can also interact with the child's standard I/O. For example, you can
+    /// read its stdout while waiting for it to exit.
+    ///
+    /// ```no_run
+    /// # use std::process::Stdio;
+    /// #
+    /// # use tokio::io::AsyncReadExt;
+    /// # use tokio::process::Command;
+    /// # use tokio::sync::oneshot::channel;
+    /// #
+    /// # use futures::future::join;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///     let (_tx, rx) = channel::<()>();
+    ///
+    ///     let mut child = Command::new("echo")
+    ///         .arg("Hello World!")
+    ///         .stdout(Stdio::piped())
+    ///         .spawn()
+    ///         .unwrap();
+    ///
+    ///     let mut stdout = child.stdout.take().expect("stdout is not captured");
+    ///     let mut buff = Vec::new();
+    ///     let wait_for_output = join(child.wait(), stdout.read_to_end(&mut buff));
+    ///
+    ///     tokio::select! {
+    ///             _ = wait_for_output => {}
+    ///             _ = rx => child.kill().await.expect("kill failed"),
+    ///     }
+    ///
+    ///     assert_eq!(buff, b"Hello World!\n");
+    /// }
```

Comment:
I think what I would recommend is that you read the output from a background task. Something along these lines:
```rs
let stdout_task = tokio::spawn(async { /* read stdout here */ });
tokio::select! {
     _ = child.wait() => {},
     _ = kill_signal => child.kill.await?,
};
let output = stdout_task.await?;
```